admin/server/organizations.go

```diff
@@ -186,6 +187,13 @@ func (s *Server) UpdateOrganization(ctx context.Context, req *adminv1.UpdateOrga
 		return nil, status.Error(codes.Internal, err.Error())
 	}
 
+	if orgName != org.Name {
```

Comment:
This will never be true – it's comparing on the result value (after the update)

Reply:
I moved this check and annotations update up so that the organization is updated in db at the end of the flow. This fixes the issue and the base concept